### PR TITLE
Loopy sprint

### PR DESCRIPTION
**Pull request by sv2518**
Let Jenkins run on Firedrake loopy sprint branch

Linked to https://github.com/firedrakeproject/tsfc/pull/244 and https://github.com/OP2/PyOP2/pull/612

Requires loopy branch https://github.com/firedrakeproject/loopy/tree/kernel_callables_v3-edit2

### CODE REVIEW

**Comment by sv2518** [comment]
The docs are failing with
```
from loopy.transform.callable import merge
68
ImportError: cannot import name 'merge'
```
I don't understand why that is happening, does someone else know why?

**Comment by wence-** [comment]
> I don't understand why that is happening, does someone else know why?

The doc build runs in the most recently built trunk container, (it should probably run a full `firedrake-update` but presently doesn't). So the loopy is out of date.

**Comment by sv2518** [comment]
I have just had a look at the config file for the actions. Should we not be checking out the branches that are specified in the Jenkinsfile with the --package-branch option?

**Comment by wence-** [comment]
> I have just had a look at the config file for the actions. Should we not be checking out the branches that are specified in the Jenkinsfile with the --package-branch option?

Probably yes. There's a broader question of how one shares "common" parts of the workflow across multiple files.

These cross-project merges are rare enough that we could not work "that hard", but it would be nice to figure something good out.

File: Jenkinsfile [hunk 1]
```diff
@@ -40,7 +40,7 @@ pipeline {
               sh 'mkdir tmp'
               dir('tmp') {
                 timestamps {
-                  sh '../scripts/firedrake-install $COMPLEX --tinyasm --disable-ssh --minimal-petsc --slepc --documentation-dependencies --install thetis --install gusto --install icepack --install irksome --install femlium --no-package-manager || (cat firedrake-install.log && /bin/false)'
+                  sh '../scripts/firedrake-install $COMPLEX --tinyasm --disable-ssh --minimal-petsc --slepc --documentation-dependencies --install thetis --install gusto --install icepack --install irksome --no-package-manager --package-branch loopy kernel_callables_v3-edit2 --package-branch PyOP2 loopy-sprint --package-branch tsfc loopy-sprint|| (cat firedrake-install.log && /bin/false)'
```
Comment (dham) [inline Jenkinsfile:43]: Drop this.

**Review by dham — APPROVED** [review]

**Pull request merged**